src/plugin.ts

```diff
@@ -250,6 +206,88 @@ function parseMessage(
   };
 }
 
+const detectJsonSpacing = (jsonString: string) => {
+  const patterns = [
+    { spacing: 1, regex: /^{\n {1}[^ ]+.*$/m },
+    { spacing: 2, regex: /^{\n {2}[^ ]+.*$/m },
+    { spacing: 3, regex: /^{\n {3}[^ ]+.*$/m },
+    { spacing: 4, regex: /^{\n {4}[^ ]+.*$/m },
+    { spacing: 6, regex: /^{\n {6}[^ ]+.*$/m },
+    { spacing: 8, regex: /^{\n {8}[^ ]+.*$/m },
+    { spacing: '\t', regex: /^{\n\t[^ ]+.*$/m }
+  ];
+
+  for (const { spacing, regex } of patterns) {
+    if (regex.test(jsonString)) {
+      return spacing;
+    }
+  }
+
+  return 2; // No matching spacing configuration found
+}
```

Comment:
@NilsJacobsen What a great idea!!!! 

If we can detect the spacing used when parsing, do we need the `serialize` setting at all?

Reply:
nope it's already removed

Reply:
Ah so the type
 
```
export type PluginSettings = {
  pathPattern: string;
  variableReferencePattern?: [string, string];
  serialize?: {
    space?: Parameters<typeof JSON.stringify>[2];
    flatten?: boolean;
  }
};
```

is outdated?

Reply:
I was wondering if we couldn't get rid of `serialize` entirely. Seems like we can store every information in the metadata on parsing.

Reply:
```js
export type PluginSettings = {
  pathPattern: string;
  variableReferencePattern?: [string, string];
};
```
That's the new/old one